src/fetchcode/package_util.py

```diff
@@ -723,3 +730,294 @@ def get_package_info(cls, gh_purl, package_name):
         "date": "2002-08-19T04:23:00",
     },
 }
+
+
+def get_cocoapods_org_url_status(purl, name, cocoapods_org_url):
+    purl_to_cocoapods_org_url_status = {}
+    cocoapods_org_url_head_request = utils.make_head_request(cocoapods_org_url)
+    cocoapods_org_url_status_code = cocoapods_org_url_head_request.status_code
+
+    if cocoapods_org_url_status_code == 404:
+        logger.error(f"cocoapods_org_url not found for {name}")
+        purl_to_cocoapods_org_url_status["return_message"] = "cocoapods_org_url_not_found"
+        return purl_to_cocoapods_org_url_status
+    elif cocoapods_org_url_status_code == 503:
+        logger.error(f"cocoapods_org_url temporarily unavailable for {name}")
+        purl_to_cocoapods_org_url_status["return_message"] = "cocoapods_org_url_temporarily_unavailable"
+        return purl_to_cocoapods_org_url_status
+    elif cocoapods_org_url_status_code == 302:
+        redirect_url = cocoapods_org_url_head_request.headers['Location']
+        redirect_message = f"The cocoapods.org URL {cocoapods_org_url} redirects to {redirect_url}"
+        logger.warning(redirect_message)
+        print(redirect_message)
```

Comment:
@keshav-space I see you've made the same comment re all the print statements that I added to accompany each of the `logger.warning()` annotations.  I'm deleting them as you suggest, but FYI I'd added them initially because when a user is running the purldb-toolkit `metadata` command and outputting to a file rather than the console, I though it useful for these warnings to be printed so the user could see them without first having to open the output file.

Deleted.

Reply:
@keshav-space Re my status on responding to your remaining comments -- I've removed the `logger.warning()` print statements and as you suggest (great idea) have replaced them with a `sys.stdout.write(line)` approach in the `construct_headers()` function in purldb-toolkit's `purlcli.py`.  (The purldb-toolkit `metadata` command has been the vehicle for creating and testing the new cocoapods support added to fetchcode's `package.py` et al. so they both need some TLC at the same time.)

That change, plus the various logging changes as well as my data restructuring ("de-nesting") of 2 of the 4 `purlcli.py` commands, led to the failure of a number of purlcli tests -- I'm attending to those now, including mocking the use of the `.log` file (shared by the fetchcode and the purldb-toolkit code) in sending data both to the console and to the "errors" and "warnings" keys in the `metadata` JSON output.

I'm turning next to the 8 remaining failed purlcli tests and will resolve these as well as attend to your remaining fetchcode/package.py comments.